### PR TITLE
Consolidate ccan sources into src/meson.build

### PR DESCRIPTION
**Pull request by martin-belanger**
The ccan directory is just used as a source directory containing
files that get built along with the other libnvme sources. There
is no point in having a meson.build in the ccan directory just to 
list the sources to be added to src/meson.build. In fact, that just 
makes it harder to see what is actually getting build. This commit 
removes the ccan/meson.build file and changes the definition of 
the "sources" in src/meson.build to include the ccan files.

Signed-off-by: Martin Belanger <martin.belanger@dell.com>

### CODE REVIEW

**Comment by martin-belanger** [comment]
Hi @igaw - This is the clean up stuff I mentioned in https://github.com/linux-nvme/libnvme/issues/174

File: src/meson.build [hunk 1]
```diff
@@ -14,19 +16,26 @@ sources += [
     'nvme/log.c',
     'nvme/tree.c',
     'nvme/util.c',
+    join_paths(source_dir, '../ccan/ccan/list/list.c'),
+    join_paths(source_dir, '../ccan/ccan/str/debug.c'),
+    join_paths(source_dir, '../ccan/ccan/str/str.c'),
```
Comment (eli-schwartz) [inline src/meson.build:21]: You don't really need to join these paths to the source dir.

What is the difference between `$PWD/../ccan/ccan/list/list.c` and `../ccan/ccan/list/list.c`? Meson processes them as files either way, and resolves them relative to current_source_dir.

There is a better reason to use it for the version-script. Meson doesn't know that that link argument is a file, so it cannot relativize it.

**Comment by igaw** [comment]
I don't know what you have against ccan/meson.build. Anyway Eli had a good point with building a private static ccan library which we just add to libnvme and the python binding library. Let's rather to this.

**Comment by martin-belanger** [comment]
Agreed. A private static ccan library would certainly be better than the current passing around of the variable `sources` between 3 separate meson.build files.

**Pull request closed without merging**